receiver/awscontainerinsightreceiver/internal/gpu/decorator_test.go

```diff
@@ -0,0 +1,233 @@
+	return m.tags[key]
+}
+
+func (m *mockGpuMetric) RemoveTag(key string) {
+	delete(m.tags, key)
+}
+
+type MockK8sDecorator struct {
+}
+
+func (m *MockK8sDecorator) Decorate(metric stores.CIMetric) stores.CIMetric {
+	return metric
+}
+
+func (m *MockK8sDecorator) Shutdown() error {
+	return nil
+}
+
+type mockNextConsumer struct {
+}
+
+func (mc *mockNextConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: true,
+	}
+}
+
+func (mc *mockNextConsumer) ConsumeMetrics(_ context.Context, md pmetric.Metrics) error {
+	return nil
+}
+
+func TestConsumeMetrics(t *testing.T) {
+	logger, _ := zap.NewDevelopment()
+	dc := &decorateConsumer{
+		containerOrchestrator: "EKS",
+		nextConsumer:          consumertest.NewNop(),
+		k8sDecorator:          &MockK8sDecorator{},
+		logger:                logger,
+	}
+	ctx := context.Background()
+
+	testcases := map[string]struct {
+		metrics     pmetric.Metrics
+		want        pmetric.Metrics
+		shouldError bool
+	}{
+		"empty": {
+			metrics:     pmetric.NewMetrics(),
+			want:        pmetric.NewMetrics(),
+			shouldError: false,
+		},
+		"unit": {
+			metrics: generateMetrics(map[string]map[string]string{
+				gpuUtil: {
+					"device": "test0",
+				},
+				gpuMemUtil: {
+					"device": "test0",
+				},
+				gpuMemTotal: {
+					"device": "test0",
+				},
+				gpuMemUsed: {
+					"device": "test0",
+				},
+				gpuPowerDraw: {
+					"device": "test0",
+				},
+				gpuTemperature: {
+					"device": "test0",
+				},
+			}),
+			want: generateMetrics(map[string]map[string]string{
+				gpuUtil: {
+					"device": "test0",
+					"Unit":   "Percent",
+				},
+				gpuMemUtil: {
+					"device": "test0",
+					"Unit":   "Percent",
+				},
+				gpuMemTotal: {
+					"device": "test0",
+					"Unit":   "Bytes",
+				},
+				gpuMemUsed: {
+					"device": "test0",
+					"Unit":   "Bytes",
+				},
+				gpuPowerDraw: {
+					"device": "test0",
+					"Unit":   "None",
+				},
+				gpuTemperature: {
+					"device": "test0",
+					"Unit":   "None",
+				},
+			}),
+			shouldError: false,
+		},
+		"noUnit": {
+			metrics: generateMetrics(map[string]map[string]string{
+				"test": {
+					"device": "test0",
+				},
+			}),
+			want: generateMetrics(map[string]map[string]string{
+				"test": {
+					"device": "test0",
+				},
+			}),
+			shouldError: false,
+		},
+		"typeUnchanged": {
+			metrics: generateMetrics(map[string]map[string]string{
+				gpuUtil: {
+					"device": "test0",
+					"Type":   "TestType",
+				},
+			}),
+			want: generateMetrics(map[string]map[string]string{
+				gpuUtil: {
+					"device": "test0",
+					"Type":   "TestType",
+					"Unit":   "Percent",
+				},
+			}),
+			shouldError: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		err := dc.ConsumeMetrics(ctx, tc.metrics)
+		if tc.shouldError {
+			assert.Error(t, err)
+			return
+		}
+		require.NoError(t, err)
+		assert.Equal(t, tc.metrics.MetricCount(), tc.want.MetricCount())
```

Comment:
super minor nit: For the `testify/assert` the order is `expected`, `actual`. It'll make it easier to understand the test failures if they're in the right order.

Reply:
oh nice catch 👍